standalone/include/psen_scan_v2_standalone/configuration/xml_configuation_parser.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2021 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#include <string>
+#include <stdexcept>
+
+#include <tinyxml2.h>
+
+#include "psen_scan_v2_standalone/configuration/zoneset_configuration.h"
+
+#ifndef PSEN_SCAN_V2_XML_CONFIGURATION_PARSER_H
+#define PSEN_SCAN_V2_XML_CONFIGURATION_PARSER_H
+
+namespace psen_scan_v2_standalone
+{
+namespace configuration
+{
+class XMLConfigurationParserException : public std::runtime_error
+{
+public:
+  XMLConfigurationParserException(const std::string& msg) : std::runtime_error(msg)
+  {
+  }
+};
+
+unsigned long ro_value_to_uint(const std::string& ro_value)
+{
+  auto string_cpy = ro_value;
+  std::swap(string_cpy[0], string_cpy[2]);
+  std::swap(string_cpy[1], string_cpy[3]);
+  return std::stoul(string_cpy, nullptr, 16);
+}
+
+/**
+ * @brief Convert string from a \<ro\> element to values.
+ *
+ * The value in a \<ro\> element is a string with length 4*N where N is the number of distance values.
+ * 4 succedding values form a set that can be transformed into the lenth in mm.
+ * The conversion of 4 characters follows the rule "abcd" -> 0xcdab
+ * Examples:
+ * "D307" -> 0x07D3 -> 2003mm
+ * "ED03" -> 0x03ED -> 1005mm
+ * "2B01" -> 0x012B -> 299mm
+ * "8913" -> 0x1389 -> 5001mm
```

Comment:
Would move this explanation to ro_value_to_uint where you really do this magic and refer to it with @ref or @see